01-node-tutorial/answers/11-fs-async.js

```diff
@@ -0,0 +1,32 @@
+const { writeFile, readFileSync } = require("fs");
+
+const path = "./temporary/fileB.txt";
+console.log("at start");
+
+writeFile(path, "This is line 1\n", (err) => {
+    console.log("at point 1");
+    if (err) {
+        console.error("Error writing line 1:", err);
+    } else {
+         writeFile(path, "This is line 2\n", { flag: 'a' }, (err) => {
+            console.log("at point 2");
+            if (err) {
+                console.error("Error writing line 2:", err);
+            } else {
+                writeFile(path, "This is line 3\n", { flag: 'a' }, (err) => {
+                    console.log("at point 3");
+                    if (err) {
+                        console.error("Error writing line 3:", err);
+                    } else {
+                        console.log("All lines written successfully.");
+                    }
+                });
+            }
+        });
+    }
+});
+
+console.log("at end");
+
+let readResult = readFileSync(path, "utf8");
+console.log(readResult);
```

Comment:
I could not execute your file.
when you hit a condition like the error you will want to raise an error or in this case log the error and break out of the loop with the `return` otherwise the program will continue to execute.

This is how it should look like:
```suggestion
const { writeFile } = require('fs')

const path = './temporary/fileB.txt'
console.log('at start')

writeFile(path, 'This is line 1\n', (err) => {
  console.log('at point 1')
  if (err) {
    console.log('Error writing line 1:', err)
    return
  }
  writeFile(path, 'This is line 2\n', { flag: 'a' }, (err) => {
    console.log('at point 2')
    if (err) {
      console.error('Error writing line 2:', err)
      return
    }
    writeFile(path, 'This is line 3\n', { flag: 'a' }, (err) => {
      console.log('at point 3')
      if (err) {
        console.error('Error writing line 3:', err)
        return
      }
      console.log('All lines written successfully.')
    })
  })
})
console.log('at end')

```